modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -58,11 +58,22 @@
 pytestmark = pytest.mark.filterwarnings(default_to_pandas_ignore_string)
 
 
-def eval_setitem(md_df, pd_df, value, col=None, loc=None):
+def eval_setitem(
+    md_df, pd_df, value=None, col=None, loc=None, md_value=None, pd_value=None
+):
+    assert value is not None or (
+        md_value is not None and pd_value is not None
+    ), "You have to either pass generic 'value' or lib-dependent values"
+
     if loc is not None:
         col = pd_df.columns[loc]
 
-    value_getter = value if callable(value) else (lambda *args, **kwargs: value)
+    if value is None:
+        value_getter = lambda src_df, *args, **kwargs: (  # noqa: E731 (do not assign a lambda expression)
+            md_value if isinstance(src_df, (pd.DataFrame, pd.Series)) else pd_value
+        )
```

Comment:
Can we pass this in to `eval_setitem` from `test_setitem_2d_insertion`? Because `eval_setitem` looks a bit confusing with `value=None`, `md_value=None`, `pd_value=None` parameters.

Reply:
I don't like the idea of putting this hard-to-read lambda in tests. I've refactored the `eval_setitem` a bit so there's no need for these triple-values. Also wrote a docstring for the function, so it shouldn't be confusing anymore.

Reply:
I still think the funciton has an ambiguity, in partitular, `value` and `pandas_value` parameters even though there is a docstring. I would suggest to use my approach for now. If we have much more tests, where we have to write such lambdas, then we could reconsider the approach.

@modin-project/modin-core may have some thoughts regarding this.

Reply:
done